### ISSUE TITLE
Mounting an encrypted filesystem

### ISSUE THREAD

**Issue by chriswue** [linuxkit/linuxkit#3245]
I spent several days on building a linuxkit image which allows me to mount an encrypted file system backed by a loop device (essentially making this work on linuxkit: http://goohackle.com/how-to-create-a-portable-encrypted-file_system-on-a-loop-file/). 

Ended up introducing two utility containers: one to run `losetup` and one to run `cryptsetup`.

Is there any interest in upstreaming this? If yes then I'll tidy it up and add documentation etc.

**Comment by justincormack** [comment]
No, I haven't seen an example, and there is definitely interest, so this would be helpful, yes.

**Comment by chriswue** [comment]
@justincormack: Currently the utility containers I have are shell scripts - looking over the repo I guess you guys would prefer the container entrypoint to be in go?

**Comment by justincormack** [comment]
Shell scripts are better than nothing! `losetup` is only a line or so of Go code to do `mount` but haven't looked at the internals of `cryptsetup` it could be a whole lot of work...
